examples/customresources/models/mynavigation/mynavigation.go

```diff
@@ -18,19 +18,47 @@ import (
 var Model = resource.NewModel("acme", "demo", "mynavigation")
 
 func init() {
-	resource.RegisterService(navigation.API, Model, resource.Registration[navigation.Service, resource.NoNativeConfig]{
+	resource.RegisterService(navigation.API, Model, resource.Registration[navigation.Service, *Config]{
 		Constructor: newNav,
 	})
 }
 
+// Config is the navigation model's config.
+type Config struct {
+	Lat  *float64 `json:"lat,omitempty"` // omitempty for a pointer to a float64 defaults to nil in golang
+	Long *float64 `json:"long,omitempty"`
+
+	// a helper function to denote that we are not checking if any attributes exist or are set to anything in particular
+	// Config structures require Validate functions if they exist in a model
```

Comment:
I'm confused what this means. I'm fairly certain you can use a resource config with no `Validate` method with no issue.

Reply:
only if you pass in no native config to the registration, Config actually looks for a Validate in its interface:
<img width="710" alt="Screenshot 2023-09-19 at 17 34 24" src="https://github.com/viamrobotics/rdk/assets/35934754/4a423ae1-dfc1-4f07-9ad3-cbc418fc56d2">

Reply:
Ah gotcha; you're totally right. I think when I wrote the original `ConfigValidator` interface it wasn't used to describe a required generic parameter of resource registration (`resource.NoNativeConfig` didn't exist either). Can we just change slightly to:
```suggestion
	// Config structs used in resource registration must implement Validate.
```